src/contracts/project.cairo

```diff
@@ -0,0 +1,527 @@
+        }
+    }
+
+    #[external(v0)]
+    impl ERC3525MetadataImpl of IERC3525Metadata<ContractState> {
+        fn contract_uri(self: @ContractState) -> felt252 {
+            let unsafe_state = ERC3525::unsafe_new_contract_state();
+            ERC3525::ERC3525MetadataImpl::contract_uri(@unsafe_state)
+        }
+
+        fn slot_uri(self: @ContractState, slot: u256) -> felt252 {
+            let unsafe_state = ERC3525::unsafe_new_contract_state();
+            ERC3525::ERC3525MetadataImpl::slot_uri(@unsafe_state, slot)
+        }
+    }
+
+    #[external(v0)]
+    impl ERC3525SlotApprovableImpl of IERC3525SlotApprovable<ContractState> {
+        fn set_approval_for_slot(
+            ref self: ContractState,
+            owner: ContractAddress,
+            slot: u256,
+            operator: ContractAddress,
+            approved: bool
+        ) {
+            let mut unsafe_state = ERC3525::unsafe_new_contract_state();
+            ERC3525::ERC3525SlotApprovableImpl::set_approval_for_slot(
+                ref unsafe_state, owner, slot, operator, approved
+            )
+        }
+
+        fn is_approved_for_slot(
+            self: @ContractState, owner: ContractAddress, slot: u256, operator: ContractAddress
+        ) -> bool {
+            let unsafe_state = ERC3525::unsafe_new_contract_state();
+            ERC3525::ERC3525SlotApprovableImpl::is_approved_for_slot(
+                @unsafe_state, owner, slot, operator
+            )
+        }
+    }
+
+    #[external(v0)]
+    impl ERC3525SlotEnumerableImpl of IERC3525SlotEnumerable<ContractState> {
+        fn slot_count(self: @ContractState) -> u256 {
+            let unsafe_state = ERC3525::unsafe_new_contract_state();
+            ERC3525::ERC3525SlotEnumerableImpl::slot_count(@unsafe_state)
+        }
+
+        fn slot_by_index(self: @ContractState, index: u256) -> u256 {
+            let unsafe_state = ERC3525::unsafe_new_contract_state();
+            ERC3525::ERC3525SlotEnumerableImpl::slot_by_index(@unsafe_state, index)
+        }
+
+        fn token_supply_in_slot(self: @ContractState, slot: u256) -> u256 {
+            let unsafe_state = ERC3525::unsafe_new_contract_state();
+            ERC3525::ERC3525SlotEnumerableImpl::token_supply_in_slot(@unsafe_state, slot)
+        }
+
+        fn token_in_slot_by_index(self: @ContractState, slot: u256, index: u256) -> u256 {
+            let unsafe_state = ERC3525::unsafe_new_contract_state();
+            ERC3525::ERC3525SlotEnumerableImpl::token_in_slot_by_index(@unsafe_state, slot, index)
+        }
+    }
+
+    // Externals
+
+    #[external(v0)]
+    impl ExternalImpl of super::IExternal<ContractState> {
+        fn total_value(self: @ContractState, slot: u256) -> u256 {
+            let unsafe_state = ERC3525::unsafe_new_contract_state();
+            ERC3525::ExternalImpl::total_value(@unsafe_state, slot)
+        }
+
+        fn mint(ref self: ContractState, to: ContractAddress, slot: u256, value: u256) -> u256 {
```

Comment:
I know this is a ~1:1 with the Cairo Zero version, yet here's the question:
Will we still have a separate Minter contract?

Reply:
I think we do in a short and middle term. It could comes later since minter are very standalone stuff in the carbon protocol, so we could setup a new coming sale and decide to switch on a miniter multi slots.